<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Editor</class>
 <widget class="QMainWindow" name="Editor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>848</width>
    <height>706</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>Fira Mono</family>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="cursor">
   <cursorShape>ArrowCursor</cursorShape>
  </property>
  <property name="windowTitle">
   <string>Make Python Editor</string>
  </property>
  <property name="statusTip">
   <string/>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">background: #f0f0f0;</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="2">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QPushButton" name="findandreplace_closebutton">
        <property name="minimumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Roboto</family>
          <pointsize>7</pointsize>
          <italic>false</italic>
         </font>
        </property>
        <property name="toolTip">
         <string extracomment="Close Find &amp; Replace">Close Find &amp; Replace | Escape</string>
        </property>
        <property name="toolTipDuration">
         <number>-1</number>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:12px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
        </property>
        <property name="text">
         <string>X</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="findText">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>27</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>27</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Roboto</family>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">QPlainTextEdit{
	color: black;
	background: white;
	border: 0px solid #0023a4;
	border-radius: 5px;
}</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="tabChangesFocus">
         <bool>true</bool>
        </property>
        <property name="undoRedoEnabled">
         <bool>false</bool>
        </property>
        <property name="lineWrapMode">
         <enum>QPlainTextEdit::NoWrap</enum>
        </property>
        <property name="centerOnScroll">
         <bool>true</bool>
        </property>
        <property name="placeholderText">
         <string>Type text to find</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="replaceText">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>27</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>27</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Roboto</family>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">QPlainTextEdit{
	color: black;
	background: white;
	border: 0px solid #0023a4;
	border-radius: 5px;
}</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="tabChangesFocus">
         <bool>true</bool>
        </property>
        <property name="undoRedoEnabled">
         <bool>false</bool>
        </property>
        <property name="lineWrapMode">
         <enum>QPlainTextEdit::NoWrap</enum>
        </property>
        <property name="centerOnScroll">
         <bool>true</bool>
        </property>
        <property name="placeholderText">
         <string>Type replacement text</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="findandreplace_findbutton">
        <property name="minimumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Roboto</family>
          <pointsize>6</pointsize>
          <italic>false</italic>
         </font>
        </property>
        <property name="toolTip">
         <string extracomment="Find">Find | Highlight Text to Find + Enter</string>
        </property>
        <property name="toolTipDuration">
         <number>-1</number>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:12px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
        </property>
        <property name="text">
         <string>üîé</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="findandreplace_replacebutton">
        <property name="minimumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Roboto</family>
          <pointsize>12</pointsize>
          <italic>false</italic>
         </font>
        </property>
        <property name="toolTip">
         <string extracomment="Replace">Replace | Highlight Replacement Text + Enter</string>
        </property>
        <property name="toolTipDuration">
         <number>-1</number>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:12px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
        </property>
        <property name="text">
         <string>‚áã</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="2">
     <widget class="QTextEdit" name="textEdit">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="font">
       <font>
        <family>Fira Mono</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QTextEdit{
	color: black;
	font: 10pt Fira Mono;
	background: white;
	border: 0px solid #0023a4;
	border-radius: 5px;
    padding: 5%;
}
QScrollBar::vertical {
        border: 0px solid white;
        background:#ebebeb;
        margin: 0px 0px 0px 0px;
		width: 10px;
}
QScrollBar::handle:vertical {
    background: #d2d2d2;
	border: 0px solid black;
	border-radius: 5px;
    min-height: 0px;
}
QScrollBar::horizontal {
        border: 0px solid white;
        background:#ebebeb;
        margin: 0px 0px 0px 0px;
		height: 10px;
}
QScrollBar::handle:horizontal {
    background: #d2d2d2;
	border: 0px solid black;
	border-radius: 5px;
    min-width: 0px;
}
QScrollBar::add-line:horizontal {
    border: 2px solid grey;
    background: #32CC99;
    width: 20px;
    subcontrol-position: right;
    subcontrol-origin: margin;
}

QScrollBar::sub-line:horizontal {
    border: 2px solid grey;
    background: #32CC99;
    width: 20px;
    subcontrol-position: left;
    subcontrol-origin: margin;
}</string>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="lineWrapMode">
       <enum>QTextEdit::NoWrap</enum>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Fira Mono'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="acceptRichText">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="2">
     <widget class="QTextEdit" name="messageText">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>27</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Fira Mono</family>
        <pointsize>7</pointsize>
        <weight>50</weight>
        <italic>false</italic>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QTextEdit{
	color: black;
	font: 7pt Fira Mono;
	background: white;
	border: 0px solid #0023a4;
	border-radius: 5px;
    padding: 2%;
}</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="undoRedoEnabled">
       <bool>false</bool>
      </property>
      <property name="lineWrapMode">
       <enum>QTextEdit::NoWrap</enum>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Fira Mono'; font-size:7pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;This is an error message&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
      </property>
     </widget>
    </item>
    <item row="0" column="0" rowspan="5">
     <widget class="QWidget" name="widget" native="true">
      <property name="minimumSize">
       <size>
        <width>100</width>
        <height>0</height>
       </size>
      </property>
      <property name="toolTipDuration">
       <number>-1</number>
      </property>
      <widget class="QWidget" name="verticalLayoutWidget">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>4</y>
         <width>92</width>
         <height>569</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="pybutton">
          <property name="minimumSize">
           <size>
            <width>90</width>
            <height>90</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>90</width>
            <height>90</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Daisy Script</family>
            <pointsize>48</pointsize>
            <italic>false</italic>
           </font>
          </property>
          <property name="toolTip">
           <string extracomment="Launch">Launch | Ctrl + Enter</string>
          </property>
          <property name="toolTipDuration">
           <number>-1</number>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton{
	color: #0023a4;
	background: transparent;
	border: 0px solid white;
	border-radius:45px;
	padding: 5px;
}

QPushButton::hover{
	border: 2px solid #94ff94;
}

QPushButton::pressed{
	color:white;
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
          </property>
          <property name="text">
           <string>py</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="5" column="0">
           <widget class="QPushButton" name="sourcebutton">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Roboto</family>
              <pointsize>11</pointsize>
              <italic>false</italic>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="Module Source">Open Module Source | Ctrl + M</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:20px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
            </property>
            <property name="text">
             <string>üëÅ</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QPushButton" name="findbutton">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Roboto</family>
              <pointsize>11</pointsize>
              <italic>false</italic>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="Find">Find | Ctrl + F</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:20px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
            </property>
            <property name="text">
             <string>üîé</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="openbutton">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Roboto</family>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="Open">Open | Ctrl + O</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:20px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
            </property>
            <property name="text">
             <string>üì§</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="savebutton">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Roboto</family>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="Save">Save | Ctrl + S</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:20px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
            </property>
            <property name="text">
             <string>üì•</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="newbutton">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Roboto</family>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="New">New | Ctrl + N</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string/>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:20px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QPushButton" name="shellbutton">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Roboto</family>
              <pointsize>11</pointsize>
              <italic>false</italic>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="Python Shell">Python Shell | Ctrl + Shift + Enter</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:20px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
            </property>
            <property name="text">
             <string>&gt;&gt;</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QPushButton" name="colorbutton">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Roboto</family>
              <pointsize>20</pointsize>
              <italic>false</italic>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="Light / Dark">Light / Dark | Ctrl + B</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:20px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
            </property>
            <property name="text">
             <string>‚òØ</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QPushButton" name="pipbutton">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <family>Roboto</family>
              <pointsize>11</pointsize>
              <italic>false</italic>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="pip">Package Installer | Ctrl+P</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
	color: white;
	background-color: #0023a4;
	border: 0px solid white;
	border-radius:20px;
	padding: 5px;
}

QPushButton::hover{
	background-color: #0C75E8;
}

QPushButton::pressed{
	border: 2px solid #94ff94;
	background-color: #0d242e;
}</string>
            </property>
            <property name="text">
             <string>üì¶</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
